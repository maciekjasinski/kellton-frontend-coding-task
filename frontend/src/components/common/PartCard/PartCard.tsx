import { MinifigPartType } from '@/types/minifig';

export const PartCard = ({ name, id, picture }: MinifigPartType) => {
  return (
    <div className="flex gap-3">
      {picture && <img src={picture} alt={name} className="size-12 object-contain" />}
      <div className="flex flex-col">
        <span className="text-sm font-bold text-info-content">{name}</span>
        <span className="text-xs text-accent">{id}</span>
      </div>
    </div>
  );
};
