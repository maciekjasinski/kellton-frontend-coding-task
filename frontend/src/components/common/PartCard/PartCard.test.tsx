import { describe, it, expect } from 'vitest';
import renderer from 'react-test-renderer';
import { PartCard } from './PartCard';

describe('PartCard', () => {
  it('should create snapshot', () => {
    const part = {
      id: 'a1',
      name: 'Test Minifig',
      picture: 'http://example.com/image.png',
    };
    const card = renderer.create(<PartCard {...part} />);
    expect(card).toMatchSnapshot();
  });
});
