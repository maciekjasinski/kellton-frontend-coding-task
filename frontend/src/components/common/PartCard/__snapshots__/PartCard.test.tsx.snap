// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PartCard > should create snapshot 1`] = `
<div
  className="flex gap-3"
>
  <img
    alt="Test Minifig"
    className="size-12 object-contain"
    src="http://example.com/image.png"
  />
  <div
    className="flex flex-col"
  >
    <span
      className="text-sm font-bold text-info-content"
    >
      Test Minifig
    </span>
    <span
      className="text-xs text-accent"
    >
      a1
    </span>
  </div>
</div>
`;
